File: consultorFinanceiro-webapi/Domain/Entities/Transaction.cs
using consultorFinanceiro_webapi.Domain.Enum;

namespace consultorFinanceiro_webapi.Domain.Entities
{
    public class Transaction
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Title { get; set; }
        public string? Description { get; set; }
        public decimal Amount { get; set; }
        public TransactionType TransactionType { get; set; }
        public DateTimeOffset CreatedDate { get; set; } = DateTimeOffset.UtcNow;
        public bool IsDeleted { get; set; } = false;

        //Categoria 
        public Guid? CategoryId { get; set; }
        public Category? Category { get; set; }

        //Usuario
        public Guid UserId { get; set; }
        public User User { get; set; }



    }
}

File: consultorFinanceiro-webapp/src/types/transaction.ts
export interface TransactionDto{
    title:string;
    description:string;
    amount:number;
    transactionType: 0 | 1;
    categoryId:string | null;
}

export interface MinimalTransaction{
    id:string;
    title:string;
    description:string;
    amount:number;
    transactionType: 0 | 1;
}

export interface ReturnTransaction{
    id:string;
    title:string;
    description:string;
    createdDate:string;
    amount:number;
    transactionType: 0 | 1;
    categoryId:string;
}